receiver/otlpreceiver/otlp_test.go

```diff
@@ -171,6 +172,118 @@ func TestGrpcGateway_endToEnd(t *testing.T) {
 	assert.EqualValues(t, got, want)
 }
 
+func TestProtoHttp(t *testing.T) {
+	addr := testutils.GetAvailableLocalAddress(t)
+
+	// Set the buffer count to 1 to make it flush the test span immediately.
+	sink := new(exportertest.SinkTraceExporter)
+	ocr, err := New(otlpReceiver, "tcp", addr, sink, nil)
+	require.NoError(t, err, "Failed to create trace receiver: %v", err)
+
+	require.NoError(t, ocr.Start(context.Background(), componenttest.NewNopHost()), "Failed to start trace receiver: %v", err)
+	defer ocr.Shutdown(context.Background())
+
+	// TODO(nilebox): make starting server deterministic
+	// Wait for the servers to start
+	<-time.After(10 * time.Millisecond)
+
+	url := fmt.Sprintf("http://%s/v1/trace", addr)
+
+	wantOtlp := []*otlptrace.ResourceSpans{
+		{
+			Resource: &otlpresource.Resource{
+				Attributes: []*otlpcommon.AttributeKeyValue{
+					{
+						Key:         conventions.AttributeHostHostname,
+						StringValue: "testHost",
+						Type:        otlpcommon.AttributeKeyValue_STRING,
+					},
+				},
+			},
+			InstrumentationLibrarySpans: []*otlptrace.InstrumentationLibrarySpans{
+				{
+					Spans: []*otlptrace.Span{
+						{
+							TraceId:           []byte{0x5B, 0x8E, 0xFF, 0xF7, 0x98, 0x3, 0x81, 0x3, 0xD2, 0x69, 0xB6, 0x33, 0x81, 0x3F, 0xC6, 0xC},
+							SpanId:            []byte{0xEE, 0xE1, 0x9B, 0x7E, 0xC3, 0xC1, 0xB1, 0x73},
+							Name:              "testSpan",
+							StartTimeUnixNano: 1544712660000000000,
+							EndTimeUnixNano:   1544712661000000000,
+							Attributes: []*otlpcommon.AttributeKeyValue{
+								{
+									Key:      "attr1",
+									Type:     otlpcommon.AttributeKeyValue_INT,
+									IntValue: 55,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
```

Comment:
Since this test largely replicates the test above that uses application/json (which is good since they're just testing different protocols), I'd refactor this so that both tests get the data from the same function/variable. And in general, maybe the tests could be refactored to share some other code as well.

Reply:
We tried to do something like this:
https://github.com/open-telemetry/opentelemetry-collector/blob/master/internal/data/testdata/trace.go

Reply:
Is this refactoring something I can do in this PR or should it be done in a separate PR?

Reply:
Separate PR is good.